postprocessing/science/compute_isochrones.py

```diff
@@ -0,0 +1,250 @@
+
+parser.add_argument('filename', help='path+filename-surface.xdmf')
+
+parser.add_argument("x", nargs=1, type=float, help="receiver x")
+parser.add_argument("y", nargs=1, type=float, help="receiver y")
+parser.add_argument("--z", nargs=1, type=float, default=([0.]), help="receiver z")
+
+parser.add_argument('--faultXdmf', type=str, metavar=('-fault.xdmf'), 
+                    help='provide path+filename-fault.xdmf; only needed when the path differs from the -surface.xdmf file')
+
+parser.add_argument('--output', choices=["numpy","xdmf","both"], default="xdmf", 
+                    help='choose the output format')
+
+parser.add_argument('--sliprateThreshold', nargs=1, default=([0.05]), metavar=('peak sliprate threshold in m/s'),
+                    help='fault cells below the treshold are assumed to not radiate waves' , type=float)
+
+parser.add_argument('--sliprateParameters', choices=["absolute","components","both"], default="both", 
+                    help="absolute: compute isochrones from peak times of absolute slip rates;"+ 
+                    " components: peak slip rate times are computed for SRs and SRd separately")
+
+parser.add_argument('--medianFilterRadius', nargs=1, default=([0.]), metavar=('radius of the median filter in m'),
+                    help='peak slip rate time of a fault cell is set to the median value of all surrounding fault cells within \
+                    a sphere with this radius' , type=float)
+# This prevents strong oscillations of peak slip rate times where different rupture fronts have similar peak slip rates 
+# and picks the time of the dominant rupture front.
+# Default = 0. turns the filter off
+# Radii between 150 m and 300 m are advised (also depending on the resolution of the fault output)
+
+parser.add_argument('--threads', nargs=1, metavar=('threads'), default=([4]),
+                    help='use multiprocessing to speed up the median filtering' ,type=int)
+
+parser.add_argument('--events', nargs=1, choices=[0,1,2], default=([0]), 
+                    help='0: xdmf file contains just one event; 1: process the first of two events; 2: process the second of two events' ,type=int) 
+# Needed for Ridgecrest output, where the xdmf file contains two events 
+
+args = parser.parse_args()
+
+
+def ComputeTriangleMidpoints(geom, connect):
+    """Generates an array with coordinates of triangle midpoints (same dimension as connect)"""
+    xyz = np.zeros_like(connect, dtype=float)
+    xyz = (1./3.)*(geom[connect[:,0],:]+geom[connect[:,1],:]+geom[connect[:,2],:])   
+    return xyz
+
+
+def GetTimeIndices(xdmfFile):
+    """Selects the correct time steps for output that contains two events"""    
+    if args.events[0] == 0:
+        return [0, xdmfFile.ReadNdt()]
+    elif args.events[0] == 1:
+        return [0, int(xdmfFile.ReadNdt()/2)]
+    else:
+        return [int(xdmfFile.ReadNdt()/2), xdmfFile.ReadNdt()]
+
+
+def LoadSliprates(sliprateParameters):
+    sliprates = []
```

Comment:
clearer with a dictionary?

Reply:
it is returned as a numpy array to later perform array operations